src/PostgREST/Query/QueryBuilder.hs

```diff
@@ -171,7 +183,9 @@ mutateRequestToQuery (Delete mainQi logicForest range ordts returnings)
     SQL.sql (returningF mainQi returnings)
 
   where
-    whereLogic = if null logicForest then mempty else " WHERE " <> intercalateSnippet " AND " (pgFmtLogicTree mainQi <$> logicForest)
+    hasEmptyBody = body == mempty
+    logicForestF = intercalateSnippet " AND " (pgFmtLogicTree mainQi <$> logicForest)
+    pgrstDeleteBodyF = SQL.sql (BS.intercalate " AND " $ (\x -> pgFmtColumn mainQi x <> " = " <> pgFmtColumn (QualifiedIdentifier mempty "pgrst_delete_body") x) <$> pkFlts)
```

Comment:
I'm not sure about the use of `columns` in the query string with DELETE.  For POST and PATCH it's used as a way to select data from the body to be inserted and updated (in the SET clause). But for DELETE it would be used to select PKs from the body to be used in the WHERE clause, which is redundant since it already filters using the PKs (as seen in this line). I can see these options:
- Do not use `columns` (but it won't have an option to use the body "as is", without JSON parsing)
- Use it and filter only PKs (will need a warning "Only PKs allowed" if other columns are selected)
- Could be used as a way to add more filters in the body.

Reply:
> Could be used as a way to add more filters in the body.

Hm, this would be useful but it would be inconsistent with bulk PATCH since `columns` couldn't do the same there.

> Use it and filter only PKs (will need a warning "Only PKs allowed" if other columns are selected)

I think this would be the best way for now. Also leaves room for an enhancement later without a breaking change.

Reply:
After deliberation on another platform, we agreed on using the third option for DELETE:

> Could be used as a way to add more filters in the body

Works for single and bulk deletes.